frame/nfts/src/features/metadata.rs

```diff
@@ -0,0 +1,173 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use frame_support::pallet_prelude::*;
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	pub(crate) fn do_set_item_metadata(
+		maybe_check_owner: Option<T::AccountId>,
+		collection: T::CollectionId,
+		item: T::ItemId,
+		data: BoundedVec<u8, T::StringLimit>,
+	) -> DispatchResult {
+		let mut collection_details =
+			Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
+
+		let item_config = Self::get_item_config(&collection, &item)?;
+		ensure!(
+			maybe_check_owner.is_none() ||
+				item_config.is_setting_enabled(ItemSetting::UnlockedMetadata),
+			Error::<T, I>::LockedItemMetadata
+		);
+
+		if let Some(check_owner) = &maybe_check_owner {
+			ensure!(check_owner == &collection_details.owner, Error::<T, I>::NoPermission);
+		}
```

Comment:
I'm a little confused about this section. So,

If a collection has no owner, then the metadata for its items is locked.
If it does have an owner, then we check an item's settings. if the setting is `UnlockedMetadata`, then it is locked?
Finally, if the collection has an owner, make sure that the provided owner is it.

Reply:
Not really. The collection always has an owner, but `maybe_check_owner.is_none()` means it's a root call, so we don't need to check that owner.
`item_config.is_setting_enabled(ItemSetting::UnlockedMetadata)` means the metadata for that item is not locked.
Given all that we have: `ensure!(is_root || metadata_unlocked, ErrorOtherwise)`